File: src/main/java/me/kuku/yuq/job/KuGouJob.java
package me.kuku.yuq.job;

import com.IceCreamQAQ.Yu.annotation.Cron;
import com.IceCreamQAQ.Yu.annotation.JobCenter;
import com.icecreamqaq.yudb.jpa.annotation.Transactional;
import me.kuku.pojo.Result;
import me.kuku.yuq.entity.KuGouEntity;
import me.kuku.yuq.entity.KuGouService;
import me.kuku.yuq.entity.QqEntity;
import me.kuku.yuq.logic.KuGouLogic;
import me.kuku.yuq.utils.BotUtils;

import javax.inject.Inject;
import java.util.List;

@JobCenter
public class KuGouJob {

	@Inject
	private KuGouService kuGouService;
	@Inject
	private KuGouLogic kuGouLogic;

	@Cron("At::d::05:42:17")
	@Transactional
	public void sign(){
		List<KuGouEntity> list = kuGouService.findAll();
		for (KuGouEntity kuGouEntity : list) {
			try {
				Result<Void> result = kuGouLogic.musicianSign(kuGouEntity);
				if (result.isFailure()){
					QqEntity qqEntity = kuGouEntity.getQqEntity();
					BotUtils.sendMessage(qqEntity, "您的酷狗音乐人自动签到失败，可能为cookie已失效，如需自动签到，请重新绑定！");
					kuGouService.delete(kuGouEntity);
				}
			}catch (Exception e){
				e.printStackTrace();
			}
		}
	}

}

File: src/main/kotlin/me/kuku/yuq/job/KuGouJob.kt
package me.kuku.yuq.job

import com.IceCreamQAQ.Yu.annotation.Cron
import com.IceCreamQAQ.Yu.annotation.JobCenter
import me.kuku.yuq.entity.KuGouService
import me.kuku.yuq.entity.Status
import me.kuku.yuq.logic.KuGouLogic
import org.springframework.stereotype.Component

@JobCenter
@Component
class KuGouJob (
    private val kuGouService: KuGouService,
    private val kuGouLogic: KuGouLogic
) {


    @Cron("03:41")
    suspend fun sign() {
        val list = kuGouService.findAll().filter { it.config.sign == Status.ON }
        for (kuGouEntity in list) {
            kuGouLogic.musicianSign(kuGouEntity)
            kuGouLogic.listenMusic(kuGouEntity)
        }
    }

}